src/main/java/com/powsybl/openloadflow/sa/AbstractSecurityAnalysis.java

```diff
@@ -244,14 +238,119 @@ SecurityAnalysisReport runSync(SecurityAnalysisParameters securityAnalysisParame
                 postContingencyResults.addAll(partitionResult.getPostContingencyResults());
                 operatorStrategyResults.addAll(partitionResult.getOperatorStrategyResults());
             }
-            result = new SecurityAnalysisResult(partitionResults.get(0).getPreContingencyResult(), postContingencyResults, operatorStrategyResults);
+            finalResult = new SecurityAnalysisResult(partitionResults.get(0).getPreContingencyResult(), postContingencyResults, operatorStrategyResults);
         }
 
         stopwatch.stop();
         LOGGER.info("Security analysis {} in {} ms", Thread.currentThread().isInterrupted() ? "cancelled" : "done",
                 stopwatch.elapsed(TimeUnit.MILLISECONDS));
 
-        return new SecurityAnalysisReport(result);
+        return new SecurityAnalysisReport(finalResult);
+    }
+
+    SecurityAnalysisResult runSimulationsOnAllComponents(LfNetworkList networks, List<PropagatedContingency> propagatedContingencies, P parameters,
+                                                         SecurityAnalysisParameters securityAnalysisParameters, List<OperatorStrategy> operatorStrategies,
+                                                         List<Action> actions, List<LimitReduction> limitReductions, LoadFlowParameters lfParameters) {
+        // run simulation on main connected component network first
+        SecurityAnalysisResult result = networks.getList().stream().filter(n -> n.getValidity().equals(LfNetwork.Validity.VALID) && n.getNumCC() == ComponentConstants.MAIN_NUM)
+                .findFirst()
+                .map(n -> runSimulations(n, propagatedContingencies, parameters, securityAnalysisParameters, operatorStrategies, actions, limitReductions))
+                .orElse(createNoResult());
+
+        List<PostContingencyResult> postContingencyResults = new ArrayList<>(result.getPostContingencyResults());
+        List<OperatorStrategyResult> operatorStrategyResults = new ArrayList<>(result.getOperatorStrategyResults());
+        NetworkResult mergedPreContingencyNetworkResult = result.getPreContingencyResult().getNetworkResult();
+        List<LimitViolation> preContingenctViolations = new ArrayList<>(result.getPreContingencyResult().getLimitViolationsResult().getLimitViolations());
+
+        if (lfParameters.getConnectedComponentMode().equals(LoadFlowParameters.ConnectedComponentMode.ALL)) {
```

Comment:
minor remark: you can use == here instead of equals since you compare enum values.